ngraph/python/tests/test_onnx/utils/onnx_backend.py

```diff
@@ -0,0 +1,131 @@
+# ******************************************************************************
+# Copyright 2018-2020 Intel Corporation
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+# ******************************************************************************
+"""
+ONNX Backend implementation.
+
+See ONNX documentation for details:
+https://github.com/onnx/onnx/blob/master/docs/Implementing%20an%20ONNX%20backend.md
+"""
+
+from typing import Any, Dict, List, Optional, Sequence, Text, Tuple
+
+import numpy
+import onnx
+from onnx.backend.base import Backend, BackendRep
+from onnx.helper import make_graph, make_model, make_tensor_value_info
+
+from ngraph.impl import Function
+from tests.runtime import get_runtime
+from tests.test_onnx.utils.onnx_helpers import import_onnx_model, np_dtype_to_tensor_type
+
+
+class OpenVinoOnnxBackendRep(BackendRep):
+    def __init__(self, ng_model_function, device="CPU"):  # type: (List[Function], str) -> None
+        super().__init__()
+        self.device = device
+        self.ng_model_function = ng_model_function
+        self.runtime = get_runtime()
+        self.computation = self.runtime.computation(ng_model_function)
+
+    def run(self, inputs, **kwargs):  # type: (Any, **Any) -> Tuple[Any, ...]
+        """Run computation on model."""
+        return self.computation(*inputs)
+
+
+class OpenVinoOnnxBackend(Backend):
```

Comment:
Maybe `Backend` should be changed to `Engine` like in the `test_case`?

Reply:
This is an implementation of the ONNX Backend interface so IMO the name is ok.